lib/internal/assert/assertion_error.js

```diff
@@ -171,51 +166,60 @@ function createErrDiff(actual, expected, operator) {
   }
 
   let printedLines = 0;
+  let identical = 0;
   const msg = kReadableOperator[operator] +
         `\n${green}+ actual${white} ${red}- expected${white}`;
   const skippedMsg = ` ${blue}...${white} Lines skipped`;
 
   for (i = 0; i < maxLines; i++) {
-    // Only extra expected lines exist
-    const cur = i - lastPos;
     if (actualLines.length < i + 1) {
-      // If the last diverging line is more than one line above and the
-      // current line is at least line three, add some of the former lines and
-      // also add dots to indicate skipped entries.
-      if (cur > 1 && i > 2) {
-        if (cur > 4) {
-          res += `\n${blue}...${white}`;
-          skipped = true;
-        } else if (cur > 3) {
+      // If more than two former lines are identical, print them. Collapse them
+      // in case more than five lines were identical.
+      if (identical > 2) {
```

Comment:
I find this form of nesting a bit confusing.

Is there a compelling reason (perf) for this nesting rather than if/else?

Reply:
(same comment for below)

Reply:
I refactored the implementation a bit to address your comment. It is now less duplication by abstracting some part. I use the nested `if` because it would otherwise mean that extra code branches have to be executed even though that's not necessary.

E.g.,

`identical === 3`

```js
if (identical > 4) {
  // ...
}
if (identical > 3) {
  // ...
}
if (identical > 2) {
  // ...
}
```